public/app/features/dashboard/components/PanelEditor/PanelEditorTabs.tsx

```diff
@@ -16,55 +16,77 @@ interface PanelEditorTabsProps {
   onChangeTab: (tab: PanelEditorTab) => void;
 }
 
-export const PanelEditorTabs: React.FC<PanelEditorTabsProps> = ({ panel, dashboard, tabs, onChangeTab }) => {
-  const styles = getPanelEditorTabsStyles();
-  const activeTab = tabs.find(item => item.active)!;
+export class PanelEditorTabs extends PureComponent<PanelEditorTabsProps> {
+  componentDidMount() {
+    const { panel } = this.props;
+    panel.on(CoreEvents.queryAdded, this.triggerForceUpdate);
+    panel.on(CoreEvents.queryRemoved, this.triggerForceUpdate);
+    panel.on(CoreEvents.transformationAdded, this.triggerForceUpdate);
+    panel.on(CoreEvents.transformationRemoved, this.triggerForceUpdate);
+  }
 
-  const getCounter = useCallback(
-    (tab: PanelEditorTab) => {
-      switch (tab.id) {
-        case PanelEditorTabId.Query:
-          return panel.targets.length;
-        case PanelEditorTabId.Alert:
-          return panel.alert ? 1 : 0;
-        case PanelEditorTabId.Transform:
-          const transformations = panel.getTransformations() ?? [];
-          return transformations.length;
-      }
+  componentWillUnmount() {
+    const { panel } = this.props;
+    panel.off(CoreEvents.queryAdded, this.triggerForceUpdate);
+    panel.off(CoreEvents.queryRemoved, this.triggerForceUpdate);
+    panel.off(CoreEvents.transformationAdded, this.triggerForceUpdate);
+    panel.off(CoreEvents.transformationRemoved, this.triggerForceUpdate);
+  }
```

Comment:
And then run the following code to unsubscribe:

```suggestion
  componentWillUnmount() {
    this.subscriptions.forEach(unsubscribe => unsubscribe());
  }
```